### PR TITLE
165 VM Improvements

### PR DESCRIPTION
**Pull request by cnalley**

### CODE REVIEW

**Review by shahedrahim — APPROVED** [review]

File: apps/vm/fsw/src/vm_Arming.cpp [hunk 1]
```diff
@@ -125,7 +125,6 @@ void VM_Arming::EnteredArmed()
     App.ActuatorArmedMsg.Armed = true;
     App.VehicleControlModeMsg.Armed = true;
     App.SetHomePosition();
-    App.SendVehicleStatusMsg();
```
Comment (mbenson1) [inline apps/vm/fsw/src/vm_Arming.cpp:128]: Is this correct?  This means the "SendVehicleStatus" message is only sent once when the state machine transitions to Armed.  We want to trend away from these single one-off messages so we can eventually support checkpoint restart.

File: apps/vm/fsw/src/vm_app.cpp [hunk 2]
```diff
@@ -519,6 +510,7 @@ int32 VM::RcvSchPipeMsg(int32 iBlocking)
                 SendVehicleManagerStateMsg();
                 SendVehicleControlModeMsg();
                 SendActuatorArmedMsg();
+                SendVehicleStatusMsg();
```
Comment (mbenson1) [inline apps/vm/fsw/src/vm_app.cpp:513]: Are the other additions of "SendVehicleStatusMsg()" necessary given it exists here?  Its better than its here so its in cyclic.

File: apps/vm/fsw/src/vm_app.cpp [hunk 3]
```diff
@@ -840,7 +825,6 @@ void VM::ProcessAppCmds(CFE_SB_Msg_t* MsgPtr)
                 HkTlm.SensorAccelMsgCount             = 0;
                 HkTlm.SafetyMsgCount                  = 0;
                 HkTlm.SensorCorrectionMsgCount        = 0;
-                HkTlm.VehicleStatusMsgCount           = 0;
```
Comment (mbenson1) [inline apps/vm/fsw/src/vm_app.cpp:843]: Are we still subscribing to this message anywhere?  If so, remove the subscription.

**Review by mbenson1 — CHANGES_REQUESTED** [review]

File: apps/vm/fsw/src/vm_Arming.cpp (hunk 1 above)
Comment (cnalley) [reply apps/vm/fsw/src/vm_Arming.cpp:128]: See line 513 below. This PR removes all calls to SendVehicleStatusMsg() during state changes and sends during cyclic wakeup.

File: apps/vm/fsw/src/vm_app.cpp (hunk 2 above)
Comment (cnalley) [reply apps/vm/fsw/src/vm_app.cpp:513]: This call should only exist here during wakeup now.

File: apps/vm/fsw/src/vm_app.cpp (hunk 3 above)
Comment (cnalley) [reply apps/vm/fsw/src/vm_app.cpp:843]: This PR also removes the subscription as VM is the sender and doesn't need to receive this message.

File: apps/vm/fsw/src/vm_Arming.cpp (hunk 1 above)
Comment (mbenson1) [reply apps/vm/fsw/src/vm_Arming.cpp:128]: Sorry.  I thought this was adding it, not removing it.  Ok.  I get it.  Thanks.

**Pull request merged**